components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -1054,61 +1028,19 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           xx  = max(1.e-20_rtype,xx)   ! set lower bound on xx to prevent division by zero
           oxx = 1._rtype/xx
 
-          if (qc_incld(i,k).ge.qsmall) &
-               qccon = (aaa*epsc*oxx+(ssat(i,k)-aaa*oxx)*odt*epsc*oxx*(1.-dexp(-dble(xx*dt))))/ab
-          if (qr_incld(i,k).ge.qsmall) &
-               qrcon = (aaa*epsr*oxx+(ssat(i,k)-aaa*oxx)*odt*epsr*oxx*(1.-dexp(-dble(xx*dt))))/ab
-
-          !for very small water contents, evaporate instantly
-          if (sup(i,k).lt.-0.001_rtype .and. qc_incld(i,k).lt.1.e-12_rtype)  qccon = -qc_incld(i,k)*odt
-          if (sup(i,k).lt.-0.001_rtype .and. qr_incld(i,k).lt.1.e-12_rtype)  qrcon = -qr_incld(i,k)*odt
-
-          if (qccon.lt.0._rtype) then
-             qcevp = -qccon
-             qccon = 0._rtype
-          endif
+          call evaporate_sublimate_precip(qc_incld(i,k),qr_incld(i,k),nr_incld(i,k),t(i,k),pres(i,k), &
+          aaa,xx,dt,ab,epsr,epsc,qv(i,k),sup(i,k),ssat(i,k),xxlv(i,k), &
+          qccon,qrcon,qcevp,qrevp,nrevp)
 
-          if (qrcon.lt.0._rtype) then
-             qrevp = -qrcon
-             nrevp = qrevp*(nr_incld(i,k)/qr_incld(i,k))
-             !nrevp = nrevp*exp(-0.2*mu_r(i,k))  !add mu dependence [Seifert (2008), neglecting size dependence]
-             qrcon = 0._rtype
-          endif
+          ! AaronDonahue: The below subroutine should be moved to within
+          ! evaporate_sublimate_precip, but doing so leads to a non-BFB
+          ! solution.
```

Comment:
I don't understand why putting the _limiter stuff at the end of evaporate_sublimate_precip isn't bfb, but I'm sure you've confirmed this is the case. You should, of course,  put _limiter stuff into evaporate_sublimate_precip as a future PR even though it's not bfb.

Reply:
I'm not sure why either, and it has driven me a bit batty :-)

I agree that the limiter stuff should go directly into the evaporate_sublimate_precip subroutine and since the PR itself will not be BFB it shouldn't be a problem to do it right.  I brought it up here hoping that someone else might ameliorate my confusion about why its not BFB when I do it the correct way.

Reply:
What if you compile with CMAKE_BUILD_TYPE=Debug or SCREAM_FPMODEL=strict? BFB-ness, particularly with CMAKE_BUILD_TYPE=Release but SCREAM_FPMODEL=strict, would then imply it's a vectorization difference.

Reply:
@AaronDonahue I'll experiment with this in P3 standalone a bit tomorrow. I think we should be able to quickly do a binary search to find which lines are breaking BFB. It will also be useful to see if P3 standalone compiled with gfortan/g++ has similar behavior.

Reply:
@ambrad , running with DEBUG compile fixed the BFB issue, so it may be a vectorization difference.

Reply:
since DEBUG compile is still BFB I have resolved this comment and consolidated the two routines into one.